src/helm/benchmark/compute_contamination_metrics.py

```diff
@@ -0,0 +1,212 @@
+        "light_instances": [
+            {
+            "input": "INPUT_TEXT1",
+            "references": [
+                "REFERENCE_TEXT_1",
+                "REFERENCE_TEXT_2"
+            ]
+            },
+            {
+            "input": "INPUT_TEXT2",
+            "references": [
+                "REFERENCE_TEXT_3",
+                "REFERENCE_TEXT_4"
+            ]
+            }
+        ]
+    }
+    """
+
+    def create_light_instance_from_dict(instance_dict: dict) -> LightInstance:
+        return LightInstance(input=instance_dict["input"], references=instance_dict["references"])
+
+    light_scenarios: List[LightScenario] = []
+    scenario_jsons = open(filename, "r").readlines()
+    for scenario_json in scenario_jsons:
+        scenario_dict: dict = json.loads(scenario_json)
+        scenario_name: str = scenario_dict["name"]
+        light_instances: List[LightInstance] = [
+            create_light_instance_from_dict(instance_dict) for instance_dict in scenario_dict["light_instances"]
+        ]
+        light_scenarios.append(LightScenario(name=scenario_name, light_instances=light_instances))
+    return light_scenarios
+
+
+def compute_instance_contamination(scenario: LightScenario, training_document: str, n: int) -> BinaryScenarioMetric:
+    def get_ngrams(text: str):
+        return list(ngrams(text.split(), n))
+
+    scenario_metric = BinaryScenarioMetric.from_scenario(scenario)
+    document_ngrams = set(get_ngrams(training_document))
+
+    for i in range(len(scenario.light_instances)):
+        instance: LightInstance = scenario.light_instances[i]
+        for ngram in get_ngrams(instance.input):
+            if ngram in document_ngrams:
+                scenario_metric.write_dirty(i, "input")
+                break
+
+        for reference in instance.references:
+            for ngram in get_ngrams(reference):
+                if ngram in document_ngrams:
+                    scenario_metric.write_dirty(i, "ref")
+                    break
+            if scenario_metric.get_bit(i, "ref") == 1:
+                break
+
+    return scenario_metric
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input-data", type=str, required=True, help="Path to your training data")
+    parser.add_argument("--scenario-data", type=str, required=True, help="Path to scenario data (benchmarking data)")
+    parser.add_argument("--output-stats", type=str, required=True, help="Path to the output file")
+    parser.add_argument(
+        "--tags",
+        type=str,
+        nargs="+",
+        help="Other tags, such as whether the input data is for pretraining or instruction tuning",
+        default=[],
+    )
+
+    args = parser.parse_args()
+
+    print(f"Loading scenario data from {args.scenario_data}")
+    scenarios = load_scenarios_from_jsonl(args.scenario_data)
+
+    print(f"Loading input data from {args.input_data}")
+    input_data = open(args.input_data, "r").read()
```

Comment:
You don't want to assume you can load this into memory

Reply:
For the file-based version, do we assume that 1 file = multiple documents with some document delimiter in between?

For the Beam / Spark based version, I assume that the `training_document` in `compute_instance_contamination` can fit in memory (i.e. typically 10-100 KB).

Reply:
I wonder if we should compute "average document size" as a sanity check.

Reply:
The file is now read document by document.